lmdeploy/serve/async_engine.py

```diff
@@ -314,6 +317,80 @@ async def gather():
         outputs = outputs[0] if need_list_wrap else outputs
         return outputs
 
+    def stream_infer(
+            self,
+            prompts: Union[List[str], str],
+            gen_config: Optional[Union[GenerationConfig,
+                                       EngineGenerationConfig]] = None,
+            do_preprocess: bool = True,
+            **kwargs):
+        """Inference a batch of prompts with stream mode.
+
+        Args:
+            prompts (List[str] | str): a batch of prompts
+            gen_config (GenerationConfig | None): a instance of
+                GenerationConfig. Default to None.
+            chat_template_config (ChatTemplateConfig | None):a instance of
+                ChatTemplateConfig. Default to None.
+            do_preprocess (bool): whether pre-process the messages. Default to
+                True, which means chat_template will be applied.
+        """
+        need_list_wrap = isinstance(prompts, str) or isinstance(
+            prompts[0], Dict)
```

Comment:
type hint for `prompts`  is `Union[List[str], str]`, but here the type is `List[Dict]`